codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -151,7 +151,11 @@ protected boolean isAggregated(HttpObject msg) throws Exception {
 
     @Override
     protected boolean isContentLengthInvalid(HttpMessage start, int maxContentLength) {
-        return getContentLength(start, -1L) > maxContentLength;
+        try {
+            return getContentLength(start, -1L) > maxContentLength;
+        } catch (final NumberFormatException e) {
+            return false;
```

Comment:
I agree, the Javadocs help but it's exactly because "invalid" has multiple meanings as you say that it's ambiguous and thus leads to the confusion as above. 😇